tests/BenchmarkDotNet.IntegrationTests/NugetReferenceTests.cs

```diff
@@ -45,16 +46,30 @@ public void RoslynToolchainDoesNotSupportNuGetPackageDependency()
 
             var supportedJob = Job.Dry.WithToolchain(toolchain);
             var supportedConfig = CreateSimpleConfig(job: supportedJob);
-            var supportedBenchmark = BenchmarkConverter.TypeToBenchmarks(typeof(WithCallToNewtonsoft), supportedConfig);
+            var supportedBenchmark = BenchmarkConverter.TypeToBenchmarks(typeof(WithCallToImmutableArray), supportedConfig);
             foreach (var benchmarkCase in supportedBenchmark.BenchmarksCases)
             {
                 Assert.Empty(toolchain.Validate(benchmarkCase, BenchmarkRunnerClean.DefaultResolver));
             }
         }
 
-        public class WithCallToNewtonsoft
+        public class WithCallToImmutableArray
         {
-            [Benchmark] public void SerializeAnonymousObject() => JsonConvert.SerializeObject(new { hello = "world", price = 1.99, now = DateTime.UtcNow });
+            private readonly double[] values;
+
+            public WithCallToImmutableArray()
+            {
+                var rand = new Random(Seed: 0);
+                values = Enumerable.Range(1, 10_000)
+                                   .Select(x => rand.NextDouble())
+                                   .ToArray();
+            }
+
+            [Benchmark]
+            public void ToImmutableArrayBenchmark()
```

Comment:
I think we can improve the test by asserting the assembly version of the package. See multiple frameworks tests for how to pass expected values to compare in the benchmark code.